### DISCUSSION TITLE
width using arbitrary values (%)

### DISCUSSION THREAD

**Discussion by Funkix6** [tailwindlabs/tailwindcss#7617] [Help] [answered]
# Hi !

## Here's my issue :

I am trying to make a progress bar using TailwindCSS and React and for some reason the arbitrary value in '%' won't work if i'm using template strings. With hard typed values it works fine though. Also it seems to only work with the number "10".

Code bellow...

### I am using :
* React version : 17.0.2
* TailwindCSS version : 3.0.23

### Progress bar component
```
    import React from "react";
    const ProgressBar = ({progress}) => {
        return(
            <div className='bg-gray-300 w-full h-2'>
                <div className={`bg-[#4a4e69] w-[${progress}%] h-full`} ></div>
            </div>
        )
    }
    export default ProgressBar;
```

### In the App.js

```
    <ProgressBar progress={20}/>
```

It's my very first issue I hope I am providing enough informations...
Thanks for taking the time to help :).

# EDIT

## How to fix : 

```
  import React from "react";
  
  const ProgressBar = ({progress}) => {
      return(
          <div className='bg-gray-300 w-full h-2'>
              <div 
                  style={{ width: `${progress}%`}}  <--- Add this
                  className={`bg-[#4a4e69] h-full`} >
              </div>
          </div>
      )
  }
  
  export default ProgressBar;

```

**Comment by adamwathan** [comment, marked as answer]
Hey! Check out the documentation on [dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) — you can't use interpolation with class names I'm afraid or Tailwind won't know what to generate since it doesn't actually execute your code.

**Reply by Funkix6** [reply]
Thanks i'll check that out and come back here if I find an alternative solution :)

**Reply by Funkix6** [reply]
I edited the topic with the easiest solution I found thanks for guiding me in the right direction @adamwathan  !
Good day to you !
